server/src/main/java/org/opensearch/identity/IdentityService.java

```diff
@@ -56,4 +64,77 @@ public Subject getSubject() {
     public TokenManager getTokenManager() {
         return identityPlugin.getTokenManager();
     }
+
+    public static IdentityService getInstance() {
+        if (instance == null) {
+            new IdentityService(Settings.EMPTY, List.of());
+        }
+        return instance;
+    }
+
+    /**
+     * Sets the application list known to the IdentityService
+     * @param applicationList A list of all principals for known applications
+     */
+    public void setApplicationList(List<Principal> applicationList) {
+        this.applicationList = List.copyOf(applicationList);
+    }
+
+    /**
+     * Returns a list of the known application principals
+     */
+    public List<Principal> getApplicationList() {
+        return List.copyOf(applicationList);
+    }
+
+    /**
+     * Returns a list of the known application principals
+     */
+    public List<String> getApplicationStrings() {
+        return applicationList.stream().map(Principal::getName).collect(Collectors.toList());
+    }
+
+
+    //TODO: Find a way to combine these
+    public Set<String> getApplicationScopes(Principal principal) {
+
+        Set<String> allScopes = ExtensionsManager.getExtensionManager().getExtensionIdMap().get(principal.getName()).getScopes();
+
+        return allScopes.stream().filter(scope -> {
+            String[] parts = scope.split("\\.");
+            if (parts.length != 3) {
+                throw new IllegalArgumentException("Invalid scope format: " + scope);
+            }
+            ScopeEnums.ScopeNamespace scopeNamespace = ScopeEnums.ScopeNamespace.fromString(parts[0]);
+            return scopeNamespace == ScopeEnums.ScopeNamespace.APPLICATION;
+        }).collect(Collectors.toSet());
+    }
+
+    public Set<String> getActionScopes(Principal principal) {
+
+        Set<String> allScopes = ExtensionsManager.getExtensionManager().getExtensionIdMap().get(principal.getName()).getScopes();
+
+        return allScopes.stream().filter(scope -> {
+            String[] parts = scope.split("\\.");
+            if (parts.length != 3) {
+                throw new IllegalArgumentException("Invalid scope format: " + scope);
+            }
+            ScopeEnums.ScopeNamespace scopeNamespace = ScopeEnums.ScopeNamespace.fromString(parts[0]);
+            return scopeNamespace == ScopeEnums.ScopeNamespace.ACTION;
```

Comment:
- Parsing scopes, should be part of the Scope class.
- Finding a scope from a string should also be part of a scope registry that knows all the possible scopes.
- Why have a different function for action scopes vs extension points?